src/components/screens/new-questionnaire-screen.tsx

```diff
@@ -114,16 +112,15 @@ const NewQuestionnaireScreen = () => {
           passScore: passScoreValue
         }
       });
+      closeAndClear();
       return createdQuestionnaire;
     } catch (error) {
-      console.error("Failed to save questionnaire", error);
-      setError("Failed to save questionnaire. Please try again.");
+      setError(`${strings.error.questionnaireSaveFailed}, ${error}`);
       setTimeout(() => {
-        setError(null);
-      }, 3000);
-    } finally {
-      closeAndClear();
+        setError(undefined);
+      }, 5000);
```

Comment:
I'm not sure the error handler should be handled in this way. If the error handler is triggered we want the user  to close the dialog manually, rather than it close after the timeout

Reply:
Jeah, it might be better. Now dialog needs to be closed manually after unsuccessful saving has accrued